src/blockdata/constants.rs

```diff
@@ -55,10 +55,16 @@ pub const SUBSIDY_HALVING_INTERVAL: u32 = 210_000;
 /// Maximum allowed value for an integer in Script.
 pub const MAX_SCRIPTNUM_VALUE: u32 = 0x80000000; // 2^31
 
-/// In Bitcoind this is insanely described as ~((u256)0 >> 32)
-pub fn max_target(_: Network) -> Uint256 {
-    Uint256::from_u64(0xFFFF).unwrap() << 208
-}
+/// The maximum possible target.
+///
+/// This value is used to calculate difficulty, which is defined as how difficult the current
+/// target makes it to find a block relative to how difficult it would be at the highest
+/// possible target (`max_target()`). Remember highest target == lowest difficulty.
+///
+/// ref: <https://en.bitcoin.it/wiki/Target>
+// In Bitcoind this is ~(u256)0 >> 32 stored as a floating-point type so it gets truncated, hence
+// the low 208 bits are all zero.
+pub fn max_target() -> Target { Target::from(0xFFFF_u64) << 208_u32 }
```

Comment:
I did not do that because I thought using `Target::MAX` is confusing if one is thinking of the target as an integer type because it would be easy to assume it was the value with all bits set like other stdlib integer types.

Reply:
I don't think it'd be confusing

Reply:
We could add the invariant that `Target` is never above `MAX` and then it'd be natural but then alost all constructors would have to start returning `Result` or truncate the value. Maybe we should call it `CONSENSUS_MAX` (and avoid having `MAX` to avoid problems).

Reply:
Breaking the "only push once a day" rule; I moved the function out of `contstants` and onto `Target`, and renamed it `max`. Making this a `const` is non-trivial for reasons discussed below (basically Rust 1.41.1).

Reply:
> non-trivial

Hmm? `Target(U256(0xFFFF_u128 << (208 - 128), 0))` Is there any problem with this?